cloud/shared/validate_variable_definitions.py

```diff
@@ -0,0 +1,76 @@
+import os
+import glob
+import json
+
+class ValidateVariableDefinitions():
+    def __init__(self, variable_definitions = {}):
+        self.variable_definitions = variable_definitions
+
+    def load_repo_variable_definitions_files(self):
+        self.variable_definitions = {}
+        cwd = os.getcwd()
+
+        definition_file_paths = glob.glob(cwd + '/cloud/**/variable_definitions.json', recursive=True)
+
+        for path in definition_file_paths:
+            with open(path, 'r') as file:
+                definitions = json.loads(file.read())
+
+                for name, definition in definitions.items():
+                    if name in self.variable_definitions:
+                        raise RuntimeError(f"Duplicate variable name: {name} at {path}")
+
+                    self.variable_definitions[name] = definition
+
+    def get_validation_errors(self):
+        all_errors = {}
+
+        for name, definition in self.variable_definitions.items():
+            definition_errors = self.validate(definition)
+
+            if len(definition_errors) > 0:
+                all_errors[name] = definition_errors
+
+        return all_errors
+
+    def validate(self, variable_definition):
+        errors = []
+
+        if not isinstance(variable_definition.get("required", None), bool):
+            errors.append("Missing 'required' field.")
+
+        if not isinstance(variable_definition.get("secret", None), bool):
+            errors.append("Missing 'secret' field.")
```

Comment:
Do you think it's better to explicitly set these fields for every variable or can we default them to False?

Reply:
I'd prefer they were explicitly set, having seen a few production incidents that were the result of misspelled key in the configuration getting ignored and the default value causing a problem.

Reply:
makes sense